US038-adding-linting-checks

Cormac-F opened the pull request:
- Installed ESLint and the relevant typescript dependencies
- Created YAML file that checks the changed files and lints them

Test file - 
<img width="363" alt="Screenshot 2023-12-05 at 3 52 06 pm" src="https://github.com/Cormac-F/infinite-harmony-ui/assets/145451997/af67e8c3-df94-409e-8997-79db09709bfa">

Output on the workflow build -
![Screenshot 2023-12-05 at 3 51 33 pm](https://github.com/Cormac-F/infinite-harmony-ui/assets/145451997/86ee559b-1a11-452a-97f3-5f69e368ca8e)

Output when pull request is created -
![Screenshot 2023-12-05 at 3 51 51 pm](https://github.com/Cormac-F/infinite-harmony-ui/assets/145451997/6c048966-0bd8-4990-9706-df09ef162d8a)

*PLEASE NOTE*
~~ESLint doesn't work locally due to the version of node being used (v14)~~
ESLint now works locally - run the following commands
nvm install 16
nvm use 16
npm run lint
GitHub Action builds using v16 to avoid multiple errors that have been noted

Commit:
git ignore

Commit:
Installed ESLint and created YAML

Commit:
changed yaml

Commit:
changed run command

Commit:
added script and changed yaml

Commit:
changed yaml

Commit:
changed node version in the yaml

Commit:
new yaml

Commit:
testing lint

Commit:
prev test change reversed

Commit:
ESLint test

Commit:
changed node version in yaml

Commit:
ESLint test 2

Commit:
removed test

Commit:
changing workflow node version

Commit:
ESLint test

Commit:
removed test

jkelso662 commented on .gitignore:
```diff
@@ -1288,3 +1288,5 @@ node_modules/yallist/README.md
 node_modules/yallist/yallist.js
 .DS_Store
 dist/views/views/index.html
+dist/
+node_modules/
```
I think L1290 is caught by `dist/` and L1288 is caught by `node_modules/` presumably

jkelso662 reviewed (changes requested):
Looks good! Maybe worth checking in with team 4 regarding getting eslint running locally. Would also like to hear if you do find out anything useful.

Noticed your PR title is missing part of the ticket code. Also you need to add Joanne as an approver as well as me. Let me know when you are ready for another review by getting github to request another review of me and Joanne. :)

jkelso662 commented:
One more thing, noticed this ticket was in the backlog thismorning. If it is being worked on, it should be `In Progress`, or if it is waiting for code review it should be in `In Review`

Commit:
gitignore update

Commit:
eslint fix

Cormac-F commented:
- Updated ESLint file
- Found local fix by using nvm to change node version to 16
- Updated PR name
- Updated git ignore

Working locally -
<img width="726" alt="Screenshot 2023-12-06 at 11 20 35 am" src="https://github.com/Cormac-F/infinite-harmony-ui/assets/145451997/9b51aff3-a11b-445d-8770-d0d520b1811d">

jkelso662 reviewed (approved):
Looks good. Fab finding the fix. Better to update the description to say about eslint working locally.

CharlesM988 reviewed (approved):
Looks great!
